beetsplug/missing.py

```diff
@@ -106,35 +111,95 @@ def __init__(self):
+            u'-a', u'--albums', dest='albums', action='store_true',
+            help=u'show missing albums for artist instead of tracks')
         self._command.parser.add_format_option()
 
     def commands(self):
         def _miss(lib, opts, args):
             self.config.set_args(opts)
-            count = self.config['count'].get()
-            total = self.config['total'].get()
-            fmt = config['format_album' if count else 'format_item'].get()
+            albms = self.config['albums'].get()
 
-            albums = lib.albums(decargs(args))
-            if total:
-                print(sum([_missing_count(a) for a in albums]))
-                return
+            helper = self._missing_albums if albms else self._missing_tracks
+            helper(lib, args)
 
-            # Default format string for count mode.
-            if count:
-                fmt += ': $missing'
+        self._command.func = _miss
+        return [self._command]
 
-            for album in albums:
-                if count:
-                    if _missing_count(album):
-                        print_(format(album, fmt))
+    def _missing_tracks(self, lib, args):
+        """Logic for determining missing tracks from each album in the library
+        (the usual case)
+        """
+        albums = lib.albums(decargs(args))
 
-                else:
-                    for item in self._missing(album):
-                        print_(format(item, fmt))
+        count = self.config['count'].get()
+        total = self.config['total'].get()
+        fmt = config['format_album' if count else 'format_item'].get()
 
-        self._command.func = _miss
-        return [self._command]
+        if total:
+            print(sum([_missing_count(a) for a in albums]))
+            return
+
+        # Default format string for count mode.
+        if count:
+            fmt += ': $missing'
+
+        for album in albums:
+            if count:
+                if _missing_count(album):
+                    print_(format(album, fmt))
+
+            else:
+                for item in self._missing(album):
+                    print_(format(item, fmt))
+
+    def _missing_albums(self, lib, args):
+        """Logic for determining missing albums from each artist in the library
+        (-a case)
+        """
+        albums = lib.albums(decargs(args))
+        # build dict mapping artist to list of their albums in library
+        albums_by_artist = {}
+        for alb in albums:
+            artist = (alb['albumartist'], alb['mb_albumartistid'])
+            artistalbs = albums_by_artist.setdefault(artist, [])
+            artistalbs.append(alb)
+            albums_by_artist[artist] = artistalbs
+
+        # build dict mapping artist to list of all albums
+        for artist, albums in albums_by_artist.items():
+            if artist[1] is None or artist[1] == "":
+                self._log.info(
+                    u"No musicbrainz ID for artist {}, skipping.",
```

Comment:
This seems like it will print "no MBID for artist Foo" if "Foo" appears once with an MBID and once without. I suppose that's a reasonable sacrifice, but it seems a little strange.